### PR TITLE
Create directory linter

### PR DESCRIPTION
**Pull request by BastianZim**
Context in  #4584

### CODE REVIEW

File: .github/workflows/correct_directory.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+         with:
+           script: |
+             github.rest.issues.createComment({
+               issue_number: context.issue.number,
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               body: 'Hi! Thanks for your contribution to conda-forge.' +
+                     '\nUnfortunately, the recipe was added directly in the _recipes_ folder without its own subfolder.\n' +
+                     'Please move the recipe file (_your package.yaml_) into a folder with the name of the package you want to submit.\n' +
+                     'Thanks!'
```
Comment (BastianZim) [inline .github/workflows/correct_directory.yml:27]: Bit confusing but not sure how to word it better.

**Comment by dopplershift** [comment]
I'm not sure how to word it better, but I like the change.

**Comment by BastianZim** [comment]
Added a second common case

**Comment by beckermr** [comment]
Do we want this in smithy?

**Comment by BastianZim** [comment]
I followed https://github.com/conda-forge/staged-recipes/issues/4584#issuecomment-350613234 for guidelines on that, and the issue is very localised to here but also happy to close the PR.

File: .github/workflows/correct_directory.yml (hunk 1 above)
Comment (beckermr) [reply .github/workflows/correct_directory.yml:27]: Maybe use paths? So you could say "recipe was added at `recipes/<feedstock-name>.yaml` and it should be at `recipes/<feedstock-name>/meta.yaml`?

Comment (BastianZim) [reply .github/workflows/correct_directory.yml:27]: Good idea. Let me update that.

Comment (BastianZim) [reply .github/workflows/correct_directory.yml]: Do we want `<feedstock-name>` or `your_package`?  `<feedstock-name>` is more consistent but maybe not something everyone knows?

**Comment by jakirkham** [comment]
> Do we want this in smithy?

Wondered about this as well. Though maybe this is less likely to show up in a feedstock since it already has the expected structure unlike staged-recipes where users need to figure it out a bit. So maybe it is ok to keep local to `staged-recipes`. We can always revisit 🙂

**Comment by jakirkham** [comment]
Thanks all! 🙏

Hopefully this helps users that bump their heads on this one

**Pull request merged**